netbench/netbench/src/duplex.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{connection::Owner, Result};
+use core::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+use futures::ready;
+use std::{io::IoSlice, mem::MaybeUninit};
+use tokio::io::{AsyncRead, AsyncWrite, ReadBuf};
+
+const READ_BUFFER_SIZE: usize = 100_000;
+const SEND_BUFFER_SIZE: usize = 100_000_000;
+
+#[derive(Debug)]
+pub struct Connection<T: AsyncRead + AsyncWrite> {
+    id: u64,
+    inner: Pin<Box<T>>,
+    stream_opened: bool,
+    send_buffer: Vec<u8>,
+    read_buffer: [MaybeUninit<u8>; READ_BUFFER_SIZE],
+}
+
+impl<T: AsyncRead + AsyncWrite> Connection<T> {
+    pub fn new(id: u64, inner: Pin<Box<T>>) -> Self {
+        Self {
+            id,
+            inner,
+            stream_opened: false,
+            send_buffer: vec![1; SEND_BUFFER_SIZE],
```

Comment:
I think it would be better to use the `s2n_quic_core::stream::testing::Data` so we're not allocating for each connection. And with that we would use vectored writing. But we can add this after if you want.

Reply:
Ah, okay. That does sound better. Adding this later sounds good to me then, if that works.